x-pack/test/api_integration/apis/endpoint/alerts.ts

```diff
@@ -66,96 +76,182 @@ export default function({ getService }: FtrProviderContext) {
+        const args: AlertListRequestQuery = {
+          order: 'desc',
+          sort: 'thread.id',
+          before: ['', '0b07c8a7-3b23-4458-9884-ee44243c2590'],
+          empty_string_is_undefined: true,
+        };
+        const { body } = await supertest
+          .get(buildUrl(null, args))
+          .set('kbn-xsrf', 'xxx')
+          .expect(200);
+        expect(body.alerts.length).to.eql(10);
+        expect(body.alerts[0].thread?.id).to.eql(undefined);
+        expect(body.alerts[9].thread?.id).to.eql(undefined);
+        expect(body.alerts[9].event?.id).to.eql('162e48bf-80fe-4235-8bb4-814984db0d83');
+      });
+
+      it('alerts api should return data using `before` on undefined primary sort values by custom sort parameter, ascending', async () => {
+        const args: AlertListRequestQuery = {
+          page_size: 25,
+          order: 'asc',
+          sort: 'thread.id',
+          before: ['', 'd4693a89-506e-4c0f-a66f-cda355de7301'],
+          empty_string_is_undefined: true,
+        };
+        const { body } = await supertest
+          .get(buildUrl(null, args))
+          .set('kbn-xsrf', 'xxx')
+          .expect(200);
+        expect(body.alerts.length).to.eql(25);
+        expect(body.alerts[0].thread?.id).to.eql(1872);
+        expect(body.alerts[24].thread?.id).to.eql(undefined);
       });
 
-      it('alerts api should return data using `after` by custom sort parameter', async () => {
+      it('alerts api should return data using `after` by custom sort parameter, descending', async () => {
+        const args: AlertListRequestQuery = {
+          sort: 'thread.id',
+          order: 'desc',
+          after: ['2180', '8362fcde-0b10-476f-97a8-8d6a43865226'],
+        };
         const { body } = await supertest
-          .get(
-            `/api/endpoint/alerts?${nextPrevPrefixDateRange}&${nextPrevPrefixPageSize}&${nextPrevPrefixOrder}&sort=thread.id&after=2180&after=8362fcde-0b10-476f-97a8-8d6a43865226`
-          )
+          .get(buildUrl(null, args))
           .set('kbn-xsrf', 'xxx')
           .expect(200);
         expect(body.alerts.length).to.eql(10);
         expect(body.alerts[0].thread.id).to.eql(1912);
       });
 
+      it('alerts api should return data using `after` on undefined primary sort values by custom sort parameter, descending', async () => {
+        const args: AlertListRequestQuery = {
+          sort: 'thread.id',
+          order: 'desc',
+          after: ['', '162e48bf-80fe-4235-8bb4-814984db0d83'],
+          empty_string_is_undefined: true,
+        };
+        const { body } = await supertest
+          .get(buildUrl(null, args))
+          .set('kbn-xsrf', 'xxx')
+          .expect(200);
+        expect(body.alerts.length).to.eql(10);
+        expect(body.alerts[0].thread?.id).to.eql(undefined);
+        expect(body.alerts[2].thread?.id).to.eql(2824);
+      });
+
```

Comment:
i've added the 4 new api tests added thru March 2nd into testrail.  woo.  thx for writing more tests